Fix spacing and font size of the message displayed when there are no domain suggestions during search

ovitrif opened the pull request:
Fixes [#15155](https://github.com/wordpress-mobile/WordPress-iOS/issues/15155) with additional enhancements.

### Description

This PR fixes a few UI issues we observed **on the error message** that is displayed when a domain search returns no domain suggestions results:

- Adds a padding to the error message to make sure it doesn't reach the horizontal edges of the screen
- Set the font size smaller to make it fit better with the style of the page.
- Fix the visibility of the message in landscape orientation when the keyboard is displayed by adjusting its vertical positioning.

The padding & the font size was discussed and agreed to constitute an improvement with @kyleaparker and it comes after a suggestion from @twstokes.



### To Test

1. Start creating a new site using the app (From My Site screen tap the ▼ icon next to the site title)
2. On the Site Picker screen tap on the ➕ menu button
3. Select **Create WordPress.com site**
4. Choose any design or tap the **SKIP** menu button
5. Type `_` or any word with non-English characters
6. **Verify** that the text message under the input is not reaching the edges of the screen & it's smaller than it originally was.
7. Rotate the device to landscape orientation
8. **Verify** that the text message is visible when the keyboard is displayed (originally it was hidden behind the keyboard)
7. Rotate the device back to portrait orientation
8. **Verify** that the vertical positioning of the text message is the same as it was at step 6 (approximately at the center when the keyboard is visible)

### Preview

| Before | After |
| --- | --- |
| <img src="https://user-images.githubusercontent.com/4588074/158990869-50644126-aee4-4048-b491-994970c06683.png" width="314"> | <img src="https://user-images.githubusercontent.com/4588074/158995454-caca2203-a38d-4da4-b6e3-39128d928cb6.png" width="314"> |

<details>
<summary><h4>Landscape</h4></summary>

| Before |
| --- |
| <img src="https://user-images.githubusercontent.com/4588074/158991223-0e94605f-81fe-42cf-b7b0-7062201b8145.png" width="628"> |


| After |
| --- |
| <img src="https://user-images.githubusercontent.com/4588074/158990518-6d92317a-21e0-4e87-a333-92c0d6ef6754.png" width="628"> |

</details>

---

### Regression Notes
1. Potential unintended areas of impact
N/A
2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A
3. What automated tests I added (or what prevented me from doing so)
N/A

#### PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

#### PR reviewer notes:

- One reviewer approval should be enough to merge this.
- No release notes are added because the main changes (PR #17985) for this will land in `19.5`

ovitrif committed:
Add spacing between error text and horizontal edges

ovitrif committed:
Set smaller error message font size

twstokes commented on WordPress/Classes/ViewRelated/Site Creation/WebAddress/WebAddressWizardContent.swift:
```diff
@@ -305,7 +305,7 @@ final class WebAddressWizardContent: CollapsableHeaderViewController {
 
         NSLayoutConstraint.activate([
-            noResultsLabel.widthAnchor.constraint(equalTo: headerView.widthAnchor),
+            noResultsLabel.widthAnchor.constraint(equalTo: headerView.widthAnchor, constant: CGFloat(-50)),
```
Everything looked great in portrait, but I noticed that in landscape the constraint didn't respect the safe area so the notch was cutting off content. If we use the TableView for the width constraint we may not have this problem. Let me know your thoughts!

**Before**

<img width="972" alt="Screen Shot 2022-03-12 at 17 36 24" src="https://user-images.githubusercontent.com/2092798/158038592-fd5c01a5-8441-494d-ad97-d0ea9fbecdfd.png">


```suggestion
            noResultsLabel.widthAnchor.constraint(equalTo: table.widthAnchor, constant: -50),
```

![image](https://user-images.githubusercontent.com/2092798/158038561-c3c32dc2-9394-4faa-8dfe-84eef5bb001f.png)

**After**

<img width="972" alt="Screen Shot 2022-03-12 at 18 31 38" src="https://user-images.githubusercontent.com/2092798/158038623-a979c80c-4e8e-4edd-9571-8158e1f1f811.png">

ovitrif replied:
Awesome, thank you for the investigation and the suggestion @twstokes 🙇 !

twstokes reviewed (commented):
👋 @ovitrif! This is looking good! I made one comment regarding the width constraint in landscape orientation when using a phone. I also tested this on iPad and everything looked great there.

**Something I noticed that's related to this view but _not_ related to your changes:**

In landscape orientation the vertical position for the label becomes ambiguous because it's based off of the CollapsableHeaderView's height. Visually there's no breakage, but I do see a warning when I do the following:

1. Run a build from Xcode and put the device (or Simulator) in landscape orientation
2. Type `_` in the "Choose a domain" view to present `noResultsLabel`
3. Click "Debug View Hierarchy" in Xcode
4. Find `noResultsLabel` in the Debug Navigator and observe the purple ⚠️ with the message "Vertical position is ambiguous."

We could follow up on this in a later issue, or tackle it here - this is not a blocker IMO.

guarani commented:
I gave this a quick test and things work well except for the issue @twstokes's raised which is pending. So I won't leave an approval yet, but LGTM 👍

ovitrif committed:
Update noResultsLabel constraint to respect the safe area

Co-authored-by: Tanner Stokes <tanner.stokes@automattic.com>

ovitrif commented:
> 👋 @ovitrif! This is looking good! I made one comment regarding the width constraint in landscape orientation when using a phone. I also tested this on iPad and everything looked great there.
> 
> We could follow up on this in a later issue, or tackle it here - this is not a blocker IMO.

Thank you @twstokes for reviewing, testing and suggesting the improvement!

I'd go for tackling it here if it's up to me to decide, since the scope of this PR is to improve the UI of the label 👍 

P.S. Many thanks for introducing me to the Xcode View Hierarchy Debugger 🙇‍♂️

ovitrif commented:
> I gave this a quick test and things work well except for the issue @twstokes's raised which is pending. So I won't leave an approval yet, but LGTM 👍

Thank you for @guarani reviewing this and checking the issue reported by Tanner!
I've committed the changes Tanner suggested since it's a good improvement 🎉

ovitrif committed:
Merge branch 'trunk' into fix/domain-invalid-query-error-ui

ovitrif committed:
Fix positioning of the error message for no results in landscape orientation

ovitrif committed:
Fix comment doc for noResultsLabel

ovitrif committed:
Merge branch 'trunk' into fix/domain-invalid-query-error-ui

ovitrif commented:
> **Something I noticed that's related to this view but _not_ related to your changes:**
> 
> In landscape orientation the vertical position for the label becomes ambiguous because it's based off of the CollapsableHeaderView's height.

Hey @twstokes!
Thank you for finding this issue and informing me about it 🙇 

I fixed it by creating the `topAnchor` constraint with `equalTo: searchHeader.bottomAnchor` instead of `headerView.bottomAnchor`. The position of both `bottomAnchor`'s is the same but the height of the searchHeader not ambiguous which removes the warning.

Upon further testing in landscape mode I found the issue that the message was not visible when the keyboard is displayed, and added a fix for it in [473798b](https://github.com/wordpress-mobile/WordPress-iOS/pull/18123/commits/473798b5db6f61c2a8e915ba8fae04025acd67b4).

Fixing the issue was not trivial especially because the heightConstraint of the `noResultsLabel` which makes the view have sort of a "virtual vertical padding" in landscape, and the only way I could fix it is with what's in the commit.

I updated the description, testing instructions and previews and for further testing of this fix.

ovitrif committed:
Merge branch 'trunk' into fix/domain-invalid-query-error-ui

twstokes commented on WordPress/Classes/ViewRelated/Site Creation/WebAddress/WebAddressWizardContent.swift:
```diff
@@ -304,11 +307,23 @@ final class WebAddressWizardContent: CollapsableHeaderViewController {
+
+    /// Sets the top inset for the noResultsLabel based on layout orientation
+    private func updateNoResultsLabelTopInset() {
+        if let constraint = view.constraints.first(where: { $0.identifier == WebAddressWizardContent.noResultsLabelTopAnchorConstraintIdentifier }) {
```
Not a blocker at all, but we could also store the constraint itself as a private optional property so we wouldn't have to search for it using the identifier.

`private var noResultsLabelTopAnchor: NSLayoutConstraint?` (optional so we don't have to worry about init)

We'd set the property after its creation on line #310, and this would turn `updateNoResultsLabelTopInset()` into:

`noResultsLabelTopAnchor?.constant = UIDevice.current.orientation.isPortrait ? Metrics.noResultsTopInset : 0`

ovitrif replied:
Yes, thank you for the suggestion!

It was my initial approach but then I changed it into what we have now, which is suboptimal (given we search for it just a few lines of code after actually defining the constraint).

ovitrif replied:
Updated according to this suggestion in [0d31bde](https://github.com/wordpress-mobile/WordPress-iOS/pull/18123/commits/0d31bde0d487035572dcd7059d744311aa0b7a49) 🙇

twstokes reviewed (approved):
Looks great @ovitrif, thank you for going above and beyond to fix the constraint issue! 🎉 I really like that we're accounting for the keyboard height now.

Tested on iPhone and iPad:

|Portrait|Landscape|
|-------|-----------|
| <img width="512" alt="iPhone Portrait" src="https://user-images.githubusercontent.com/2092798/159021038-cce23529-b656-4b4a-8f77-a79d293c57ba.png"> | <img width="898" alt="iPhone Landscape" src="https://user-images.githubusercontent.com/2092798/159021079-69d4a620-4e7a-4a9e-92cd-651e320d0bd2.png"> |
| <img width="1036" alt="iPad Portrait" src="https://user-images.githubusercontent.com/2092798/159021318-574e2d52-f131-4820-a2da-fcc645540d13.png"> | <img width="1376" alt="iPad Landscape" src="https://user-images.githubusercontent.com/2092798/159021349-dfcdc41b-dba2-4287-b2b0-05bb194e3e6f.png"> |

ovitrif committed:
Use reference for noResultsLabelTopAnchor constraint

twstokes reviewed (approved):
LGTM @ovitrif! :shipit:
